cli/cmd/uninstall/uninstall.go

```diff
@@ -345,13 +367,60 @@ func (c *Command) uninstallHelmRelease(releaseName, namespace, releaseType strin
 		}
 	}
 
+	// Delete any custom resources managed by Consul. If they cannot be deleted,
+	// patch the finalizers to be empty on each one.
+	if releaseType == common.ReleaseTypeConsul {
+		uiLogger("Deleting custom resources managed by Consul")
+		crds, err := c.fetchCustomResourceDefinitions()
+		if err != nil {
+			return fmt.Errorf("unable to fetch Custom Resource Definitions for Consul deployment: %v", err)
+		}
+		kindToResource := mapCRKindToResourceName(crds)
+		err = backoff.Retry(func() error {
+			crs, err := c.fetchCustomResources(crds)
+			if err != nil {
+				return err
+			}
+			if len(crs) == 0 {
+				return nil
+			}
+
+			if err = c.deleteCustomResources(crs, kindToResource); err != nil {
+				return err
+			}
+
+			crs, err = c.fetchCustomResources(crds)
+			if err != nil {
+				return err
+			}
+			if len(crs) != 0 {
+				return common.NewDeletionError(fmt.Sprintf("%d custom resources remain after deletion request. Retrying deletion", len(crs)))
+			}
```

Comment:
🤔 I think this is the only thing that needs to be in the retry block, right? We don't need to retry deletion, we just need to give some time to check that they are deleted if I remember correctly. I suppose it would work either way, but just wondering if there's some reason why we needed to do that.

Reply:
You are totally right. I had this idea that maybe if the CRs change after the initial deletion request, this would help with that, but this request moves so quickly, I don't think it's a case we will reasonably move into. It's also not really how people tend to use CRs unless they have an operator that is creating CRs for them.